phpBB/phpbb/storage/controller/controller.php

```diff
@@ -0,0 +1,185 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\storage\controller;
+
+use phpbb\cache\service;
+use phpbb\db\driver\driver_interface;
+use phpbb\exception\http_exception;
+use phpbb\storage\storage;
+use Symfony\Component\HttpFoundation\Request as symfony_request;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+
+/**
+ * Generic controller for storage
+ */
+class controller
+{
+	/** @var service */
+	protected $cache;
+
+	/** @var driver_interface */
+	protected $db;
+
+	/** @var storage */
+	protected $storage;
+
+	/** @var StreamedResponse */
+	protected $response;
+
+	/** @var symfony_request */
+	protected $symfony_request;
+
+	/**
+	 * Constructor
+	 *
+	 * @param service				$cache
+	 * @param driver_interface		$db
+	 * @param storage				$storage
+	 * @param symfony_request		$symfony_request
+	 */
+	public function __construct(service $cache, driver_interface $db, storage $storage, symfony_request $symfony_request)
+	{
+		$this->cache = $cache;
+		$this->db = $db;
+		$this->storage = $storage;
+		$this->symfony_request = $symfony_request;
+		$this->response = new StreamedResponse();
```

Comment:
Looking at this again, this probably should be moved to the handle method, not injected into the controller